src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpTraceHelper.cs

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Security;
+
+namespace System.Net.Http
+{
+    internal static class WinHttpTraceHelper
+    {
+        private const string WinHtpTraceEnvironmentVariable = "WINHTTPHANDLER_TRACE";
+        private static bool s_TraceEnabled;
+
+        static WinHttpTraceHelper()
+        {
+            string env;
+            try
+            {
+                env = Environment.GetEnvironmentVariable(WinHtpTraceEnvironmentVariable);
+            }
+            catch (SecurityException)
+            {
+                env = null;
+            }
+            
+            if (!string.IsNullOrEmpty(env))
+            {
+                s_TraceEnabled = true;
+            }
+            else
+            {
+                s_TraceEnabled = false;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsTraceEnabled()
+        {
+            if (s_TraceEnabled)
+            {
+                return true;
+            }
+            
+            return false;
+        }
+
+        public static void TraceStatusCallback(string message, IntPtr handle, bool invokeCallback, uint status)
+        {
+            if (!IsTraceEnabled())
```

Comment:
Noted for future PR.  The current trace logic now does check for "IsEnabled" before computing any strings, etc.